src/liballoc/collections/btree/map.rs

```diff
@@ -4,9 +4,10 @@ use core::fmt::Debug;
 use core::hash::{Hash, Hasher};
 use core::iter::{FromIterator, FusedIterator, Peekable};
 use core::marker::PhantomData;
+use core::mem::{self, ManuallyDrop};
```

Comment:
Cc @Mark-Simulacrum @ssomers for BTreeMap changes

Reply:
I hardly understand ManuallyDrop, but isn't it a good fit for btree::node::Root? I don't really understand Root's comment "Note that this does not have a destructor," to begin with. Doesn't Rust generate a destructor if you don't implement Drop? So the comment should say: "Note that we (hopefully) do not invoke the generated destructor, because we need to clean up the entire tree and we do that in a fancy and dangerous way while iterating".

Reply:
I suppose that comment just means that there is no `Drop` impl that would clean up the data behind the raw pointers in `Root` -- usually one would expect such an `impl` to exist.

Reply:
Okay. But is that something more formally expressed by making Root.node ManuallyDrop? I should just try it and learn.

Reply:
No, putting a raw pointer (or Unique) behind manually drop is pointless.